elasticdl/python/data/parallel_table_reader.py

```diff
@@ -0,0 +1,111 @@
+from multiprocessing import Process, Queue
+
+from odps import ODPS
+
+
+class ParallelTableReader(object):
+    def __init__(
+        self,
+        access_id,
+        access_key,
+        project,
+        endpoint,
+        table,
+        partition,
+        columns,
+        batch_size,
+        num_parallel_processes,
+        transform_fn,
+    ):
+        self._odps_table = ODPS(
+            access_id, access_key, project, endpoint
+        ).get_table(table)
+        self._partition = partition
+        self._columns = columns
+        self._batch_size = batch_size
+
+        self._num_parallel_processes = num_parallel_processes
+        self._transform_fn = transform_fn
+
+        self._result_queue = Queue()
+        self._index_queues = []
+        self._workers = []
+
+        self._shards = []
+        self._shard_idx = 0
+        self._worker_idx = 0
+
+        for i in range(self._num_parallel_processes):
+            index_queue = Queue()
+            self._index_queues.append(index_queue)
+
+            p = Process(target=self._worker_loop, args=(i,))
+            p.daemon = True
+            p.start()
```

Comment:
Do we miss the code snippet to call p.join()?

Reply:
Done